src/internal/FluencePeer.ts

```diff
@@ -0,0 +1,350 @@
+    //async addConnection(relays: Array<ConnectionSpec>): Promise<void> {}
+
+    // TODO:: implement api alongside with multi-relay implementation
+    //async removeConnections(relays: Array<ConnectionSpec>): Promise<void> {}
+
+    /**
+     * Creates a new Fluence Peer instance. Does not start the workflows.
+     * In order to work with the Peer it has to be initialized with the `init` method
+     */
+    constructor() {}
+
+    /**
+     * Get the information about Fluence Peer connections
+     */
+    get connectionInfo(): ConnectionInfo {
+        const isConnected = this._connection?.isConnected();
+        return {
+            isConnected: isConnected,
+            selfPeerId: this._selfPeerId,
+            connectedRelays: this._relayPeerId ? [this._relayPeerId] : [],
+        };
+    }
+
+    /**
+     * Initializes the peer: starts the Aqua VM, initializes the default call service handlers
+     * and (optionally) connect to the Fluence network
+     * @param config - object specifying peer configuration
+     */
+    async init(config?: PeerConfig): Promise<void> {
+        let peerId;
+        const peerIdOrSeed = config?.peerId;
+        if (!peerIdOrSeed) {
+            peerId = await randomPeerId();
+        } else if (isPeerId(peerIdOrSeed)) {
+            // keep unchanged
+            peerId = peerIdOrSeed;
+        } else {
+            // peerId is string, therefore seed
+            peerId = await peerIdFromEd25519SK(peerIdOrSeed);
+        }
+        this._selfPeerIdFull = peerId;
+
+        await this._initAirInterpreter(config?.avmLogLevel || 'off');
+
+        this._callServiceHandler = makeDefaultClientHandler();
+
+        if (config?.connectTo) {
+            let connectTo;
+            if (Array.isArray(config!.connectTo)) {
+                connectTo = config!.connectTo;
+            } else {
+                connectTo = [config!.connectTo];
+            }
+
+            let theAddress: ConnectionSpec;
+            let fromNode = (connectTo[0] as any).multiaddr;
+            if (fromNode) {
+                theAddress = new MA(fromNode);
+            } else {
+                theAddress = new MA(connectTo[0] as string);
+            }
+
+            await this._connect(theAddress);
+        }
+    }
+
+    /**
+     * Uninitializes the peer: stops all the underltying workflows, stops the Aqua VM
+     * and disconnects from the Fluence network
+     */
+    async uninit() {
+        await this._disconnect();
+        this._callServiceHandler = null;
+    }
+
+    /**
+     * Get the default Fluence peer instance. The default peer is used automatically in all the functions generated
+     * by the Aqua compiler if not specified otherwise.
+     */
+    static get default(): FluencePeer {
+        return this._default;
+    }
+
+    // internal api
+
+    /**
+     * Does not intended to be used manually. Subject to change
+     */
+    get internals() {
```

Comment:
> Does not intended to be used manually. Subject to change

Why isn't it `private` then?

Reply:
Because it is used in the code generated by the Aqua compiler